### PR TITLE
Fix #4532

### LINKED ISSUE #4532: Top level results get filtered out if nested connections with interfaces are empty when sorting
Top level results get filtered if a query is the connections to an interface are empty and these are sorted.

This happens on GraphQL 4.x
The following instructions to reproduce are for the coming up GraphQL 5.0.0 version

Given the following schema:

```graphql
type Inventory {
    id: ID
    children: [Scenario!]!
        @relationship(type: "HasChildren", properties: "InventoryChildRelation", direction: OUT)
}

interface Scenario {
    id: ID
}

type Image implements Scenario {
    id: ID
}

type Video implements Scenario {
    id: ID
}

type InventoryChildRelation @relationshipProperties {
    order: Int
}
```

And the following data:

```cypher
CREATE(i:Inventory {id: "i1"})
CREATE(i)-[:HasChildren { order: 3 }]->(c1:Image { id: "c3"})
CREATE(i)-[:HasChildren { order: 1 }]->(c2:Video { id: "c1"})
CREATE(i)-[:HasChildren { order: 2 }]->(c3:Video { id: "c2"})

CREATE(:Inventory {id: "i2"})
```

The following query:

```graphql
query {
  inventories {
      id
      childrenConnection(sort: { edge: { order: ASC } }) {
          edges {
              properties {
                  order
              }
              node {
                  id
              }
          }
      }
  }
}
```

Should return both `Inventory` items, however, the second one gets filtered out. This does not happen with connections to normal types or if sort is removed

### PR DESCRIPTION
**Pull request by angrykoala**
# Description

Fixes a cardinality problem in the Cypher, where some nodes would be filtered out if the nested relationship is sorted

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Thanks for the Neo4j GraphQL Toolbox updates.

The Neo4j GraphQL Toolbox has now been torn down - reopening this PR will republish it.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3404 | 3404 | 33 | 37 | 1134 |
|  aggregations.TopLevelAggregateWithMultipleFields | 6802 | 6802 | 46 | 48 | 1134 |
|  aggregations.NestedAggregation | 16544 | 16544 | 57 | 81 | 2174 |
|  aggregations.AggregationWithWhere | 11970 | 11970 | 37 | 56 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116478 | 22116478 | 2156 | 2193 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116478 | 22116478 | 2247 | 2224 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9701 | 9701 | 30 | 42 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9928 | 9928 | 36 | 55 | 2174 |
|  aggregations.DeeplyNestedCount | 11062184 | 11062184 | 2905 | 3203 | 2008534 |
|  aggregations.InterfacesAggregations | 6242 | 6242 | 41 | 98 | 2080 |
|  aggregations.InterfacesAggregationsWithTwoFields | 11444 | 11444 | 64 | 80 | 2080 |
|  batch-create.BatchCreate | 4200 | 4200 | 120 | 140 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 91 | 55 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 180 | 161 | 3003 |
|  connections.Connection | 14049 | 14049 | 95 | 70 | 2174 |
|  connections.NestedConnection | 41068 | 41068 | 117 | 124 | 4516 |
|  create.SimpleMutation | 7 | 7 | 48 | 55 | 1 |
|  cypher-directive.TopLevelMutationDirective | 1135 | 1135 | 25 | 22 | 1134 |
|  delete.SimpleDelete | 19401 | 19401 | 212 | 203 | 1040 |
|  delete.NestedDeleteInUpdate | 19165 | 19165 | 190 | 223 | 1179 |
|  2871.NestedRelationshipFilter | 23073 | 23046 | 70 | 56 | 4429 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 35 | 39 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22891 | 22891 | 66 | 97 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 34 | 42 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 46 | 67 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 18 | 21 | 1040 |
|  query.SimpleQueryWithRelationship | 16129 | 16129 | 36 | 41 | 2174 |
|  query.QueryWhere | 9675 | 9688 | 32 | 40 | 2161 |
|  query.SimpleQueryWithNestedWhere | 9842 | 9855 | 44 | 48 | 2161 |
|  query.Nested | 10095502 | 10095502 | 6831 | 6924 | 2008534 |
|  query.NestedWithFilter | 10074351 | 10074351 | 6791 | 6960 | 2004000 |
|  query.OrFilterOnRelationships | 42799 | 41385 | 132 | 159 | 1754 |
|  query.OrFilterOnRelationshipsAndNested | 35874 | 34246 | 178 | 223 | 1754 |
|  query.QueryWithNestedIn | 13454 | 13245 | 45 | 64 | 1245 |
|  query.NestedConnectionWhere | 9801 | 9801 | 45 | 64 | 2174 |
|  query.DeeplyNestedConnectionWhere | 9850 | 9852 | 64 | 85 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 18758 | 18776 | 131 | 147 | 1586 |
|  query.NestedWithRelationshipSingleFilters | 3865 | 3865 | 141 | 186 | 1134 |
|  query.Fulltext | 64 | 64 | 26 | 36 | 16 |
|  query.FulltextWithNestedQuery | 558 | 558 | 34 | 49 | 84 |
|  sorting-and-cypher.TopLevelSortWithCypher | 14059 | 14059 | 40 | 59 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypher | 14059 | 14059 | 58 | 69 | 2174 |
|  sorting-and-cypher.TopLevelSortWithCypherWithNested | 14184 | 14200 | 54 | 59 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithCypherWithNested | 14184 | 14200 | 105 | 114 | 2174 |
|  sorting-and-cypher.TopLevelSortWithExpensiveCypher | 14967 | 14629 | 93 | 129 | 2174 |
|  sorting-and-cypher.TopLevelConnectionSortWithExpensiveCypher | 14493 | 14504 | 99 | 122 | 2174 |
|  sorting.SortMultipleTypes | 3498 | 3498 | 79 | 83 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 14437 | 14448 | 91 | 103 | 2174 |
|  sorting.SortOnNestedFields | 14049 | 14049 | 42 | 47 | 2174 |
|  sorting.SortDeeplyNestedFields | 43148 | 43148 | 85 | 95 | 4516 |
|  sorting.ConnectionWithSort | 3281 | 3281 | 62 | 69 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 49 | 55 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 47 | 51 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 246 | 252 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 220 | 258 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 86 | 110 | 2002 |

</details>
        
Old Schema Generation: 26.553s
Schema Generation: 25.795s
Old Subgraph Schema Generation: 28.849s
Subgraph Schema Generation: 29.384s

File: packages/graphql/tests/integration/issues/4532.int.test.ts [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+import { cleanNodes } from "../../utils/clean-nodes";
+import { UniqueType } from "../../utils/graphql-types";
+import { default as Neo4j } from "../neo4j";
+
+describe("https://github.com/neo4j/graphql/issues/4532", () => {
+    let driver: Driver;
+    let neo4j: Neo4j;
+    let neo4jGraphql: Neo4jGraphQL;
+
+    describe("order-by relationship property should", () => {
```
Comment (mjfwebb) [inline packages/graphql/tests/integration/issues/4532.int.test.ts]: should?

Comment (angrykoala) [reply packages/graphql/tests/integration/issues/4532.int.test.ts]: :sweat_smile: 
```suggestion
    describe("order-by relationship property", () => {
```

**Pull request merged**